content/articles/run-length-encoding/index.md

```diff
@@ -0,0 +1,150 @@
+                break
+        encoded_string = encoded_string + str(count) + ch
+        i = j + 1
+    return encoded_string
+```
+
+The code is for encoding a message.
+It is a function `encode_message()` with a parameter `message` for the string to be encoded.
+
+The variable, `encoded_string`, is for storing the encoded string. The `while` loop initializes the count to 1 for storing the human-readable count, 1, 2, 3, 4,...., instead of starting from 0. 
+
+The innermost `while` loop checks if the character at the current index is the same as the character at the next index. If the characters are the same, the count is incremented to 1. If they are not the same, the counter is stopped, then the count and the character is concatenated to the `encoded_string` variable. This goes on until all the characters are encoded.
+
+The `encoded_string` is returned after the function has finished executing.
+
+The next function is for decoding the encoded message. Create a new cell and paste this code.
+
+```python
+def decode(our_message):
+    decoded_message = ""
+    i=0
+    j=0
+    while (i <= len(our_message)-1):
+        run_count = int(our_message[i])
+        run_word = our_message[i + 1]
+        for j in range(run_count):
+            decoded_message = decoded_message+run_word
+            j = j + 1
+        i = i + 2
+    return decoded_message
+```
+
+THe function `decode()` takes in the parameter `our_message` for the encoded massage.
+
+We have a variable `decoded_message` for storing the decoded string.
+
+The `while` loop is for splitting the encoded message into respective counts, `run_count`, and the character, `run_word`, until it is finished. It contains a `for` loop for displaying the character multiple times specified by the count(`run_count`) to form a string. The string is then concatenated with the `decoded_message` variable. This continues until the decoded string is displayed in full.
+
+Finally, we write the method for displaying what we have written. Again, open a new cell and paste this.
+
+```python
+def display():
+    our_message = "AuuBBBCCCCCCcccccCCCCCCCCCA"
+    encoded_message=encode_message(our_message)
+    decoded_message=decode_message(encoded_message)
+    print(encoded_message)
+    print(decoded_message)
+```
+
+The function, `display()`, has the string to be encoded and decoded.
+
+We then call the two methods for the process. We pass in the original string to the `encode_message()` function. After encoding, it is stored in the `encoded_message` variable. 
+
+The encoded message, `encoded_message`, is passed to the `decode_message` function for decoding. Thereafter, the decoded message is stored in the `decoded_message` variable.
+
+We then print the two messages. The output will be as shown below:
+
+```bash
+1A2u3B6C5c9C1A
+AuuBBBCCCCCCcccccCCCCCCCCCA
+```
+
+And that's it. We have looked at what is compression, the types of compression, Run Length Encoding, and implementing it in Python.
```

Comment:
It was an abrupt conclusion.

While concluding, always give a summary of what is being taught, the key takeaways from this article. Add a separate section `Conclusion`.